src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -100,7 +102,9 @@ internal sealed partial class LocalRewriter : BoundTreeRewriterWithStackGuard
                 {
                     dynamicAnalysisSpans = dynamicInstrumenter.DynamicAnalysisSpans;
                 }
-
+#if DEBUG
+                LocalRewritingValidator.Validate(loweredStatement);
```

Comment:
>LocalRewritingValidator.Validate(loweredStatement); [](start = 16, length = 51)

I am concerned that this introduces a new source of stack overflow exception and only for debug. Some test will be testing different code path debug vs. release. Perhaps we should catch and swallow the internal stack overflow exception thrown by this validator.

Reply:
Added try/catch inside that method

---
In reply to: [162696640](https://github.com/dotnet/roslyn/pull/23961#discussion_r162696640) [](ancestors = 162696640)